Clean up AppDelegate

mdoukmak opened the pull request:
- Remove UIKeyCommand constants
- Create UIKeyCommand extension to hold constants

mdoukmak committed:
Clean up AppDelegate
- Remove UIKeyCommand constants
- Create UIKeyCommand extension to hold constants

olivaresf commented on Zavala/UIKeyCommand+Zavala.swift:
```diff
@@ -0,0 +1,185 @@
+import Foundation
+import UIKit
+
+let appDelegate = UIApplication.shared.delegate as! AppDelegate
```
- Move this inside the extension and mark it fileprivate.
- Add a comment explaining that we're only reaching out to the AppDelegate to declare UIKeyCommand selectors.

In general, accessing the AppDelegate singleton is frowned upon. It's actually very common for new developers to declare properties in the AppDelegate as if it were a global object. Since there's no way for us to declare the selectors without access to the AppDelegate, this is correct usage but a comment that adds context would be a good idea.

mdoukmak replied:
Thanks for the feedback! Ideally I would create another object that would encapsulate this 'link' between UIKeyCommand and AppDelegate but for this simple change I guess this will do.

olivaresf commented on Zavala/UIKeyCommand+Zavala.swift:
```diff
@@ -0,0 +1,185 @@
+
+    static let sync = UIKeyCommand(title: L10n.sync,
+                                   action: #selector(appDelegate.syncCommand(_:)),
+                            input: "r",
```
Indentation is incorrect.

Small details like this matter in the long run think of indentation as punctuation in a regular sentence for example this sentence has no punctuation that doesn't mean you can't read it it just means that you need to exert a bit more mental power to process it.

mdoukmak replied:
Good catch. Fixed!

mdoukmak committed:
Minor updates
- Added comment on appDelegate usage, and moved inside extension.
- Fixed indentation

mdoukmak committed:
Move extension file to Utility directory

mdoukmak committed:
Fix UIKeyCommand references for macOS Catalyst

olivaresf commented on Zavala/AppDelegate.swift:
```diff
@@ -629,47 +463,47 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
-										children: [expandAllInOutlineCommand, expandAllCommand, expandCommand, collapseAllInOutlineCommand, collapseAllCommand, collapseCommand])
-		builder.insertChild(expandCollapseMenu, atStartOfMenu: .view)
-		let toggleFilterOutlineMenu = UIMenu(title: "", options: .displayInline, children: [toggleOutlineFilterCommand, toggleOutlineHideNotesCommand])
+        // View Menu
```
When this happens it means that your indentation is different than that of the project. Make sure it's the same (tabs or spaces). Once you fix it, these changes should mostly disappear.

olivaresf commented on Zavala/Utility/UIKeyCommand+Zavala.swift:
```diff
@@ -0,0 +1,187 @@
+import Foundation
+import UIKit
+
+
```
Extra space here.

olivaresf commented on Zavala/Utility/UIKeyCommand+Zavala.swift:
```diff
@@ -0,0 +1,187 @@
+
+extension UIKeyCommand {
+    
+    // We need appDelegate here only to populate the #selector actions for each command. Ideally we would not reach over to appDelegate from here, but there is no way to reach these methods without doing so.
```
✅
